vertx-opentracing/src/main/java/io/vertx/tracing/opentracing/OpenTracingTracer.java

```diff
@@ -48,88 +43,94 @@ public OpenTracingTracer(boolean closeTracer, Tracer tracer) {
   }
 
   @Override
-  public <R> Scope receiveRequest(Context context, R request, String operation, Iterable<Map.Entry<String, String>> headers, TagExtractor<R> tagExtractor) {
+  public <R> Span receiveRequest(Context context, R request, String operation,
+    Iterable<Map.Entry<String, String>> headers, TagExtractor<R> tagExtractor) {
     SpanContext sc = tracer.extract(Format.Builtin.HTTP_HEADERS, new TextMap() {
       @Override
       public Iterator<Map.Entry<String, String>> iterator() {
         return headers.iterator();
       }
+
       @Override
       public void put(String key, String value) {
         throw new UnsupportedOperationException();
       }
     });
-    Tracer.SpanBuilder builder = tracer.buildSpan(operation);
-    if (sc != null) {
-      builder = builder.asChildOf(sc);
+    try (CapturedContext ignored = VertxContextScopeManager.withContext(context)) {
+      Tracer.SpanBuilder builder = tracer.buildSpan(operation);
+      if (sc != null) {
+        builder = builder.asChildOf(sc);
+      }
+      Span span = builder
+        .ignoreActiveSpan()
+        .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_SERVER)
+        .withTag(Tags.COMPONENT.getKey(), "vertx").start();
+      tracer.activateSpan(span);
```

Comment:
The scope should not be activated here. There is no method which returns the scope anymore, so if you do not pass it around there will be a leak